DATALAYER_INTEGRATION.md

```diff
@@ -0,0 +1,281 @@
+
+The data layer client library is included by default by the Page component. As there are other ways to include this library (e.g. through Adobe Launch), it might be needed to prevent its inclusion through the Page component.
+
+To prevent the data layer client library from being included by the Page component:
+1. Create the following structure below the `/conf` node:
+   `<sling:configRef-path-at-jcr:content-of-form>/sling:configs/com.adobe.cq.wcm.core.components.internal.DataLayerConfig`
+2. Add the `skipClientlibInclude` boolean property and set it to `true`.
+
+## Data Layer State Structure
+
+When the data layer is enabled, the javascript `adobeDataLayer` object is available on the page and is populated with the components and their properties that are used on the page.
+
+The data layer state (returned by calling `adobeDataLayer.getState()`) is an object with two objects (`page` and `component`). All the components are stored below the `component` object as a flat structure. The structure looks as follows:
+```
+{
+  "page": {
+    "page-id": {
+      "key1": "value1,
+      "key2": "value2,
+    }
+  },
+  "component": {
+    "component-id1": {
+      "key1": "value1,
+      "key2": "value2,
+    },
+    "component-id2": {
+      "key1": "value1,
+      "key2": "value2,
+    },
+    ...
+  }
+}
+```
+
+Calling `adobeDataLayer.getState()` in the browser console will return e.g.:
+
+```
+{
+    "page": {
+        "page-12266a293b": {
+            "@type": "forms-components-examples/components/page",
+            "repo:modifyDate": "2023-04-24T07:08:11Z",
+            "dc:title": "af2",
+            "xdm:template": "/conf/core-components-examples/settings/wcm/templates/af-blank-v2",
+            "xdm:language": "en",
+            "xdm:tags": [],
+            "repo:path": "/content/forms/af/af2.html"
+        }
+    },
+    "component": {
+        "textinput-1c7bc238a6": {
+            "dc:title": "Name",
+            "repo:modifyDate": "2023-04-24T07:10:10Z",
+            "@type": "forms-components-examples/components/form/textinput",
+            "fieldType": "text-input",
+            "parentId": "page-12266a293b"
+        },
+        "numberinput-3bf4611d5a": {
+            "dc:title": "Phone Number",
+            "repo:modifyDate": "2023-04-24T07:10:26Z",
+            "@type": "forms-components-examples/components/form/numberinput",
+            "fieldType": "number-input",
+            "parentId": "page-12266a293b"
+        },
+        "button-961d435b34": {
+            "dc:title": "Submit Form",
+            "repo:modifyDate": "2023-04-24T07:10:37Z",
+            "@type": "forms-components-examples/components/form/button",
+            "fieldType": "button",
+            "parentId": "page-12266a293b"
+        }
+    }
+}
+```
+
+## Form Components supporting the Data Layer
+
+The following table shows the components supporting the data layer:
+
+Form Components | Data Layer Support
```

Comment:
pls use this list: https://github.com/adobe/aem-core-forms-components#adaptive-form